lib/mongo/index/view.rb

```diff
@@ -251,6 +254,10 @@ def send_initial_query(server)
 
       def with_generated_names(models)
         models.dup.each do |model|
+          if model[:key].key?(:partial_filter_expression)
+            model[:partial_filter_expression] = model[:key][:partial_filter_expression]
+            model[:key].delete(:partial_filter_expression)
+          end
```

Comment:
When I comment out those 4 lines and run: client[:test].indexes.create_one({"x":1, partial_filter_expression:{"a":{"$lte":1.5}}})
I get an error:
"bad index key pattern { x: 1, partial_filter_expression: { a: { $lte: 1.5 } } }: Index keys cannot be Objects or Arrays." }

Am I passing in the arguments incorrectly?